### DISCUSSION TITLE
Modal to edit model attribute value

### DISCUSSION THREAD

**Discussion by c-gross** [orchidsoftware/platform#2460] [Q&A] [answered]
Hi,

I would like to use a modal to give the user the option to edit a field of a model object. In my case it would a json field so the form field would be matrix.

Has anyone did something like that already and has a working code example? The docs are to basic. I guess I have to use asyncParameters() at the ModalToggle call to pass a parameter but no clue how to access that parameter in the modal / matrix field since there is nothing in the docs about it beside the method be mentioned.

Do I need to call async() on the modal too? Sounds like thats used to load data but not sure, could also be for saving?
Where I had to put the async method to be put into? The same screen class like the rest of the modal is placed or should that be an own class? Will that async method receive the params send via asyncParameters()?

**Comment by tabuna** [comment, marked as answer]
Look in the already installed application for the user list screen. 
There is an example of quick user editing in a modal window (By clicking on email)

**Comment by c-gross** [comment]
Oh nice. Thats the first thing I try to do. If there is already something like that implemented or in the example code but I never clicked on the email in the user list.
